### PR TITLE
MWPW-140270: Adding the fg sync workflow

### PR DESCRIPTION
**Pull request by JasonHowellSlavin**
* Adds workflow action for FG sync

Resolves: [MWPW-140270](https://jira.corp.adobe.com/browse/MWPW-140270)

**Test URLs:**
- Before: https://main--bacom--adobecom.hlx.live/?martech=off
- After: https://add-bac om-pink-action--bacom--adobecom.hlx.live/?martech=off

### CODE REVIEW

**Review by sukamat — APPROVED** [review]

File: .github/workflows/fg-sync-repos.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          repositories: "bacom-pink"
+
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Clone Floodgate Repository
+        run: |
+          git clone https://github.com/adobecom/bacom-pink.git ../bacom-pink
```
Comment (Brandon32) [inline .github/workflows/fg-sync-repos.yml:26]: bacom-pink?

File: .github/workflows/fg-sync-repos.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Commit and Push Changes to Floodgate Repository
+        run: |
+          cd ../bacom-pink
+          git config user.email "$FG_SYNC_BOT_EMAIL"
+          git config user.name "$FG_SYNC_BOT_NAME"
+          git status          
+          git remote set-url origin https://oauth2:$GITHUB_TOKEN@github.com/adobecom/bacom-pink.git
+          git remote -v
+          git add .
+          git commit -m "Syncing bacom to bacom-pink"
```
Comment (Brandon32) [inline .github/workflows/fg-sync-repos.yml:41]: Couldn't we just add a remote and push to it?

**Comment by Brandon32** [comment]
So the idea is to have a new repo with a separate mount point?

**Comment by JasonHowellSlavin** [comment]
> So the idea is to have a new repo with a separate mount point?

Correct. Once the Akamai cookie is set up, you can point to the same production URL but it will point to bacom-pink instead of bacom, and will point to files mounted in the bacom-pink content repository. This allows for new and even limited time content to be accessed, and then stakeholders can decide which portions of this content get "promoted" to bacom to enter the bacom content repository. 

See: 

- https://wiki.corp.adobe.com/display/WP4/Floodgate+Git+Repo+Setup
- https://wiki.corp.adobe.com/display/WP4/Milo+Floodgate+Architecture

**Review by meganthecoder — APPROVED** [review]

**Review by Brandon32 — APPROVED** [review]

File: .github/workflows/fg-sync-repos.yml (hunk 2 above)
Comment (JasonHowellSlavin) [reply .github/workflows/fg-sync-repos.yml:41]: Not sure. Thoughts @sukamat ?

**Pull request merged**